MekHQ/src/mekhq/campaign/market/unitMarket/AtBMonthlyUnitMarket.java

```diff
@@ -150,8 +239,19 @@ public void addOffers(final Campaign campaign, final int num, UnitMarketType mar
             final Collection<MissionRole> missionRoles = new ArrayList<>();
             if (unitType == UnitType.TANK) {
                 movementModes.addAll(IUnitGenerator.MIXED_TANK_VTOL);
-                missionRoles.add(MissionRole.MIXED_ARTILLERY);
+
+                // should a special unit type be picked? This allows us to force MissionRole that would otherwise be filtered out
+                if (Compute.randomInt(30) == 0) {

```

Comment:
How did we arrive at "1 in 30 tanks should be artillery or support"?

Reply:
That was just the value that 'felt right' during testing.

There is an argument to be made for making this a campaign option, but given how roles now work it's not possible to use the old method of generation, so I had to pick a value.

After Stucco's role overhaul, the `MIXED_ARTILLERY` MissionRole now filters out anything that isn't mixed artillery (which just happens filter out all vehicles). Previously this role was unique in that it deliberately didn't filter anything out, it just told mhq to include mixed artillery in addition to other units.

Not including some kind of arbitrary system to force the unit market to add artillery (and support) units results in those units never being added to the unit market.  Potentially this is due to filtering occurring elsewhere in the code, but I couldn't find anything that would indicate this taking place.

Reply:
Fair enough, we can always revisit it in a subsequent PR.